ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -413,7 +414,15 @@ private void handleStreamedQuery(String query)
           public void run() {
             for (long rowsRead = 0; keepReading(rowsRead) && queryStream.hasNext(); rowsRead++) {
               try {
-                terminal.printStreamedRow(queryStream.next());
+                StreamedRow row = queryStream.next();
+                terminal.printStreamedRow(row);
+                if (row.getErrorMessage() != null) {
+                  // got an error in the stream, which means we have reached the end.
+                  // the stream interface that queryStream uses isn't smart enough to figure
+                  // out when the socket is closed, so just break here since we know there will
+                  // be nothing more to read.
+                  break;
```

Comment:
Shall we print out the error message? In case of limit we were showing `LIMIT reached for the partition.`

Reply:
Its printed by printStreamedRow (line 418)